EIPS/eip-fhe.md

```diff
@@ -0,0 +1,55 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Homomorphically encrypted storage
+    Author:	silur <huohuli@protonmail.ch>
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-11-07
+
+
+## Simple Summary
+Storages could include a layer where data is stored and handled by the same opcodes using fully homomorphic encryption, allowing to store and evaluate any boolean logic on confidental data while
+preserving concensus on modifications.
+
+## Abstract
+Byzantium upgrade made zk-snarks possible but as we know with zk-snarks, data doesn't actually move only the validity of the commitment provided on the data.
+Fully homomorphic encryption allows anyone to bootstrap a "storage-key" for their data which allows any endparty to execute mutually agreed boolean circuits on
+encrypted data, without decrypting it, including searching and indexing too. This allows secure storage and operations on healthcare records or customer databases, enabling services easy compliance with EU GDPR. Concensus about data modification will still be transparent for storage keys are stored in the contracts.
+
+## Motivation
+A very common problem among people who wants to migrate or enter their business to blockchains is that they cannot put confidential data into a public ledger,
+although they still need to overcome distrust between different types of customers. Uploading critical business data into a blockchain where competitors or unauthorized personnel can scrape and analyze it is not an option so in all cases these parties stay with the centralized non-transparent model.
+This EIP issues this problem with enabling distribution of existing infrastructures on the blockchain, without exposing sensitive data into publicity.
+
+## Specification
+
+Contract storages will have a subset (or extension?) dedicated for encrypted data with a map of storage-keys on which can operate on a specified offset range.
```

Comment:
If you modify `SSTORE` and `SLOAD`, the plain text would be visible to the Ethereum Virtual Machine, whose execution is visible to everybody.

I'm confused who would see the plaintext, and who should perform the encryption.

Reply:
plaintext never enters the chain SSTORE/SLOAD will most likely have another 2 pairs to indicate FHE context